### PR TITLE
[DEV-2682] transaction controller refactoring

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
- [ ] add migration

File: src/subdomains/core/history/services/history.service.ts [hunk 1]
```diff
@@ -62,15 +62,14 @@ export class HistoryService {
     const all =
       query.buy == null && query.sell == null && query.staking == null && query.ref == null && query.lm == null;
 
-    const buyCryptos =
-      (all || query.buy) && (await this.buyCryptoService.getUserTransactions(user.id, query.from, query.to));
-    const buyFiats =
-      (all || query.sell) && (await this.buyFiatService.getUserTransactions(user.id, query.from, query.to));
+    const transaction = await this.transactionService.getTransactionsForUser(user.id, query.from, query.to);
+
```
Comment (davidleomay) [inline src/subdomains/core/history/services/history.service.ts:66]: `const transactions =  ...`

**Review by davidleomay — APPROVED** [review]

File: src/subdomains/supporting/payment/services/transaction.service.ts [hunk 2]
```diff
@@ -68,14 +68,31 @@ export class TransactionService {
...
+        buyFiat: { sell: true, cryptoInput: true, bankTx: true },
+        refReward: true,
+      },
+    });
+  }
+
+  async getTransactionsForUser(userId: number, from = new Date(0), to = new Date()): Promise<Transaction[]> {
+    return this.repo.find({
+      where: { user: { id: userId }, type: Not(IsNull()), created: Between(from, to) },
+      relations: {
```
Comment (davidleomay) [inline src/subdomains/supporting/payment/services/transaction.service.ts:86]: Before we used the output date, maybe we should use the updated field here?

File: src/subdomains/generic/kyc/services/kyc-client.service.ts [hunk 3]
```diff
@@ -80,10 +80,9 @@ export class KycClientService {
 
   // --- HELPER METHODS --- //
   private async toPaymentDto(userId: number, dateFrom: Date, dateTo: Date): Promise<PaymentWebhookData[]> {
-    const txList = [
-      ...(await this.buyCryptoService.getUserTransactions(userId, dateFrom, dateTo)),
-      ...(await this.buyFiatService.getUserTransactions(userId, dateFrom, dateTo)),
-    ];
+    const txList = await this.transactionService
+      .getTransactionsForUser(userId, dateFrom, dateTo)
+      .then((txs) => txs?.filter((t) => t.buyCrypto || t.buyFiat)?.map((t) => t.buyCrypto || t.buyFiat));
```
Comment (davidleomay) [inline src/subdomains/generic/kyc/services/kyc-client.service.ts:85]: `txs` cannot be null -> remove both `?`

**Review by davidleomay — APPROVED** [review]

**Pull request merged**